libbpf-tools/mountsnoop.c

```diff
@@ -277,6 +288,14 @@ int main(int argc, char **argv)
 
 	obj->rodata->target_pid = target_pid;
 
+	use_ringbuf = probe_ringbuf();
+	if (use_ringbuf) {
+		bpf_map__set_autocreate(obj->maps.events_perfbuf, false);
+		bpf_map__set_autocreate(obj->maps.heap, false);
+	} else {
+		bpf_map__set_autocreate(obj->maps.events_ringbuf, false);
+	}
+
```

Comment:
I think this should be hidden inside bpf_buffer__new(), why each app should do this over and over again?